Fix field types and add "jump to active field" text button.

FadhlanR opened the pull request:
![Screenshot 2023-10-10 at 21 11 52](https://github.com/cardstack/boxel/assets/12637010/3df3a6b2-645d-4f5c-bd25-489053bcab9f)

FadhlanR committed:
Add collection type for linksToMany field

FadhlanR committed:
Use different tag to query

jurgenwerk reviewed (commented):
In the design, it says: "Link, Collection" while this PR says "Collection, Linked"

<img width="385" alt="image" src="https://github.com/cardstack/boxel/assets/273660/ab4e43a4-1ea4-4a4e-ab32-d6f56850d434">

jurgenwerk commented:
Another thing since this PR is a follow up to https://github.com/cardstack/boxel/pull/718

Did you forget the "Jump to active field definition ⤴️" text button? I can't see it in this video here https://github.com/cardstack/boxel/pull/718#issuecomment-1754826105

FadhlanR committed:
Update types order

FadhlanR committed:
Add jump to active field

FadhlanR commented:
@jurgenwerk I've updated the order of the type and added the "Jump to active field definition ⤴️" text button. 


https://github.com/cardstack/boxel/assets/12637010/aaf75d66-d1d1-47e6-b33a-aa99afbfb02e

FadhlanR committed:
lint fixes

FadhlanR committed:
Fix test

jurgenwerk commented on packages/host/app/components/operator-mode/card-schema-editor.gts:
```diff
@@ -136,7 +137,10 @@ export default class CardSchemaEditor extends Component<Signature> {
-        gap: var(--boxel-sp-xxs);
+      }
+
+      .right > *:not(:nth-last-child(3)) {
```
Why 3? Looks oddly specific - will this break when new elements are added?

FadhlanR replied:
I removed it and add the margin to a specific class. [d398580](https://github.com/cardstack/boxel/pull/722/commits/d398580b8f7b9a7843782a9ee4933a43667c6c03)

jurgenwerk commented on packages/host/app/components/operator-mode/card-schema-editor.gts:
```diff
@@ -302,79 +314,95 @@ export default class CardSchemaEditor extends Component<Signature> {
-                        {{svgJar
-                          'warning'
+                  {{#if (this.isOverridden field)}}
+                    {{! template-lint-disable no-invalid-interactive}}
```
This linting rule is here for a reason - to prevent accessibility issues (probably a screen reader is not going to know to tell the user that they can click on a `span`)

FadhlanR replied:
Agree, I used button. [d398580](https://github.com/cardstack/boxel/pull/722/commits/d398580b8f7b9a7843782a9ee4933a43667c6c03)

FadhlanR committed:
Update some styling

FadhlanR committed:
Merge branch 'main' into add-collection-type-for-linksToMany
